cmd/contact-exporter/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"gopkg.in/gorp.v1"
+
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/sa"
+)
+
+type contactExporter struct {
+	log   blog.Logger
+	dbMap *gorp.DbMap
+	clk   clock.Clock
+}
+
+func (c contactExporter) findContacts() ([]string, error) {
+	var contactsJSON []string
+	var contacts []string
+
+	_, err := c.dbMap.Select(
+		&contactsJSON,
+		`
+			SELECT DISTINCT (contact) AS active_contact
+			FROM registrations
+			WHERE contact != 'null' AND
+				id IN (
+					SELECT registrationID
+					FROM certificates
+					WHERE expires >= :expireCutoff
+				)
+			;
+			`,
+		map[string]interface{}{
+			"expireCutoff": c.clk.Now(),
+		})
+	if err != nil {
+		c.log.AuditErr(fmt.Sprintf("contact-exporter: Error finding contacts: %s", err))
+		return nil, err
+	}
+
+	for _, contactRaw := range contactsJSON {
+		var contactList []string
+		err := json.Unmarshal([]byte(contactRaw), &contactList)
+		if err != nil {
+			c.log.AuditErr(
+				fmt.Sprintf("contact-exporter: couldn't unmarshal contact JSON %#v: %s\n",
+					contactRaw, err))
+			continue
+		}
+		for _, entry := range contactList {
+			// Only include email addresses
+			if strings.HasPrefix(entry, "mailto:") {
+				contacts = append(contacts, strings.TrimPrefix(entry, "mailto:"))
```

Comment:
Would be better to stick the contacts into a map and then serialize the map. That way we get de-duplication. Probably want to sort the map after serializing so that we get relatively stable output.